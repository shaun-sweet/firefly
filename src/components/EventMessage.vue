<template>
  <q-item class="event row justify-between">
    <div class="device">{{device}}</div>
    <div class="state">{{state}}</div>
    <div class="timestamp">{{timestamp}}</div>
  </q-item>
</template>

<script>
export default {
  props: ['device', 'state', 'timestamp'],
  data () {
    return {}
  }
}
</script>

<style lang="stylus" scoped>
</style>
