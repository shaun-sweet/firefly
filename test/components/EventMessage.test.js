import Vue from 'vue';
import { shallow } from 'vue-test-utils';
import EventMessage from '@/EventMessage.vue';

describe('EventMessage', () => {
  let wrapper;
  let propsData;
  beforeEach(() => {
    propsData = {
      device: "test device",
      state: "on",
      timestamp: 0
    }
    wrapper = shallow(EventMessage,{
      propsData
    });
  })

  it('should render EventMessage correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  });
  it('should render the props', () => {
    const divs = wrapper.findAll('div')
    const props = ['device', 'state', 'timestamp']
    const numDivs = divs.length - 1
    for(let i = 0; i <= numDivs; i++){
      for(let index in props){
        let prop = props[index]
        let div = divs.at(i)
        if(div.hasClass(prop)){
          expect(div.text()).toBe(propsData[prop].toString())
          break;
        }
      }
    }
  })

})
