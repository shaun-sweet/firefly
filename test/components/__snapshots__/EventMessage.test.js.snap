// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventMessage should render EventMessage correctly 1`] = `
<q-item
  class="event row justify-between"
>
  <div
    class="device"
  >
    test device
  </div>
   
  <div
    class="state"
  >
    on
  </div>
   
  <div
    class="timestamp"
  >
    0
  </div>
</q-item>
`;
